The test that verifies that unit test failures result in job failures is correctly-failing because the `go tool cover` command's exit status is now last/reported, not the `go test` command.

```
[19:06:23.152] [shipyard]$ go tool cover -func unit_coverage.out
total:	(statements)	0.0%
[19:06:23.307] [shipyard]$ exit 0
Successfully remade target file 'unit'.
[19:06:23.310] [shipyard]$ make --debug=b unit
Error: Process completed with exit code 1.
```

--- a/scripts/shared/unit_test.sh
+++ b/scripts/shared/unit_test.sh
@@ -54,7 +54,10 @@ for module in "${modules[@]}"; do
 
         echo "Running tests in ${packages[*]}"
         [ "${ARCH}" == "amd64" ] && race=-race
-        ${GO:-go} test -v ${race} -cover "${packages[@]}" -ginkgo.v -ginkgo.trace -ginkgo.reportPassed -ginkgo.reportFile junit.xml "$@"
+        # It's important that the `go test` command's exit status is reported from this () block.
+        # Can't be one command (with -cover). Need detailed -coverprofile for Sonar and summary to console.
+        ${GO:-go} test -v ${race} -coverprofile unit_coverage.out "${packages[@]}" -ginkgo.v -ginkgo.trace -ginkgo.reportPassed -ginkgo.reportFile junit.xml "$@" && \
+        go tool cover -func unit_coverage.out
     ) || result=1
 done
 

